### PR TITLE
feat: Add Alert controller 🚨

### PR DESCRIPTION
**Pull request by ziadali22**
**Alert controller with Builder pattern**
how to use : 

```
  UIAlertController.Builder()
      .withTitle("Test")
      .withMessage("you need to enable Gps")
      .addDefaultActionWithTitle("action1")
      .addActionWithTitle("action2", style: .default) { _ in
          print("action2")
      }
      .addCancelActionWithTitle("Cancel") { _ in
          print("cancel")
      }
      .show(in: self)
```

  or: 

 `UIAlertController.okAlert(withTitle: "Test2", withMessage: "you need to enable Gps !!", viewController: self)`

### CODE REVIEW

**Review by HebaMad — APPROVED** [review]

File: UIDokan/Sources/UIDokan/CustomTemplates/AlertController.swift [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                alert.addAction(action)
+            }
+
+            return alert
+        }
+
+        private func unwrapString(_ value: String?) -> String? {
+            guard let value = value else { return nil }
+            return value
+        }
```
Comment (ahmdmhasn) [inline UIDokan/Sources/UIDokan/CustomTemplates/AlertController.swift]: ?

File: UIDokan/Sources/UIDokan/CustomTemplates/AlertController.swift [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        private var sourceRect: CGRect?
+
+        init() {}
+
+        func preferredStyle(_ style: UIAlertController.Style) -> Builder {
+            preferredStyle = style
+            return self
+        }
+
+        func withTitle(_ title: String?) -> Builder {
```
Comment (ahmdmhasn) [inline UIDokan/Sources/UIDokan/CustomTemplates/AlertController.swift]:
```suggestion
        func title(_ title: String?) -> Builder {
```

File: UIDokan/Sources/UIDokan/CustomTemplates/AlertController.swift [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+            self.title = unwrapString(title)
+            return self
+        }
+
+        func withMessage(_ message: String?) -> Builder {
```
Comment (ahmdmhasn) [inline UIDokan/Sources/UIDokan/CustomTemplates/AlertController.swift]:
```suggestion
        func message(_ message: String?) -> Builder {
```

File: UIDokan/Sources/UIDokan/CustomTemplates/AlertController.swift [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+        func addCancelActionWithTitle(_ title: String, handler: ((UIAlertAction) -> Void)? = nil) -> Builder {
+            return addActionWithTitle(title, style: .cancel, handler: handler)
+        }
+
+        func addActionWithTitle(_ title: String, style: UIAlertAction.Style, handler: ((UIAlertAction) -> Void)?) -> Builder {
+            let action = UIAlertAction(title: NSLocalizedString(title, comment: ""), style: style, handler: handler)
+            actions.append(action)
+            return self
+        }
```
Comment (ahmdmhasn) [inline UIDokan/Sources/UIDokan/CustomTemplates/AlertController.swift]: Let's separate actions in an extension

File: UIDokan/Sources/UIDokan/CustomTemplates/AlertController.swift [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+            self.title = unwrapString(title)
+            return self
+        }
+
+        func withMessage(_ message: String?) -> Builder {
+            self.message = unwrapString(message)
+            return self
+        }
+
+        func withPopoverSourceView(_ view: UIView?) -> Builder {
```
Comment (ahmdmhasn) [inline UIDokan/Sources/UIDokan/CustomTemplates/AlertController.swift]:
```suggestion
        func popoverSourceView(_ view: UIView?) -> Builder {
```

File: UIDokan/Sources/UIDokan/CustomTemplates/AlertController.swift [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+ * This extension implements a Builder pattern to use with the UIAlertViewController.
+ * Sample:
+ * UIAlertController.Builder()
+ *    .withTitle("this_is_a_key_from_localizable.strings")
+ *    .withMessage("This is a normal message")
+ *    .addOkAction()
+ *    .show(in: UIViewController) OR .build() to get back the UIAlertController directly
+ */
+
+extension UIAlertController {
```
Comment (ahmdmhasn) [inline UIDokan/Sources/UIDokan/CustomTemplates/AlertController.swift]:
```suggestion
/// This extension implements a Builder pattern to use with the UIAlertViewController.
/// Example:
/// ```
/// UIAlertController.Builder()
///    .withTitle("this_is_a_key_from_localizable.strings")
///    .withMessage("This is a normal message")
///    .addOkAction()
///    .show(in: UIViewController) OR .build() to get back the UIAlertController directly
/// ```
extension UIAlertController {
```

File: UIDokan/Sources/UIDokan/CustomTemplates/AlertController.swift [hunk 7]
```diff
@@ -0,0 +1,115 @@
+//
+//  File.swift
+//
+//
+//  Created by ziad on 04/08/2022.
```
Comment (ahmdmhasn) [inline UIDokan/Sources/UIDokan/CustomTemplates/AlertController.swift]: File should be moved to UIDokan+Extensions+UIAlertContoroller+Builder.swift

**Review by ahmdmhasn — COMMENTED** [review]
🚀

**Review by ahmednaagy — APPROVED** [review]

**Review by ahmdmhasn — APPROVED** [review]
🎉

**Review by oradwanomar — APPROVED** [review]

**Pull request merged**